int32 nCookies
pos[] cookiesPos
